src/main/java/nl/tudelft/contextproject/gui/CameraFeedsController.java

```diff
@@ -0,0 +1,245 @@
+     */
+    private void addChoiceListener(ChoiceBox<Camera> cb) {
+        cb.getSelectionModel().selectedItemProperty().addListener((obs, oldV, newV) -> {
+            
+            LiveStreamHandler streamHandler;
+            
+            if (cb.equals(camChoiceOne)) {
+                streamHandler = leftStreamHandler;
+            } else {
+                streamHandler = rightStreamHandler;
+            }
+            
+            VBox outerBox = (VBox) cb.getParent();
+            VBox innerBox = (VBox) outerBox.getChildren().get(1);
+            ImageView oldStream = (ImageView) innerBox.getChildren().get(0);
+            
+            if (newV.toString().equals("None")) {
+                blackView(oldStream, streamHandler);
+                return;
+            }
+           
+            String newStream = newV.getConnection().getStreamLink();           
+            
+            updateStream(newStream, oldStream, streamHandler);
+        });
+    }
+    
+    private void addStreamListeners() {
+        streamBoxOne.widthProperty().addListener((observable, oldValue, newValue) -> {
+            if (leftStreamHandler != null) {
+                fitImageViewSize(newValue.floatValue(), 
+                        (float) streamBoxOne.getHeight(), 
+                            (ImageView) streamBoxOne.getChildren().get(0), leftStreamHandler);
+            }
+        });
+
+        streamBoxTwo.widthProperty().addListener((observable, oldValue, newValue) -> {
+            if (rightStreamHandler != null) {
+                fitImageViewSize(newValue.floatValue(), 
+                        (float) streamBoxTwo.getHeight(), 
+                            (ImageView) streamBoxTwo.getChildren().get(0), rightStreamHandler);
+            }
+        });
+        
+        streamBoxOne.heightProperty().addListener((observable, oldValue, newValue) -> {
+            if (leftStreamHandler != null) {
+                fitImageViewSize((float) streamBoxOne.getWidth(), 
+                        newValue.floatValue(), 
+                            (ImageView) streamBoxOne.getChildren().get(0), leftStreamHandler);
+            }
+        });
+        
+        streamBoxTwo.heightProperty().addListener((observable, oldValue, newValue) -> {
+            if (rightStreamHandler != null) {
+                fitImageViewSize((float) streamBoxTwo.getWidth(), 
+                        newValue.floatValue(), 
+                            (ImageView) streamBoxTwo.getChildren().get(0), rightStreamHandler);
+            }
+        });
+    }
+    
+    /**
+     * Updates the camera stream to the stream referenced by the specified link.
+     * @param streamLink the link to the video stream to be played next.
+     */
+    private void updateStream(String streamLink, ImageView imageView, LiveStreamHandler streamHandler) {
+        VBox vBox = (VBox) imageView.getParent();
+        
+        if (streamHandler != null) {
```

Comment:
Here you avoid a null pointer exception but still use streamhandler out of the if statement respectively in lines 154 and lines 166. It would be better to let the if statement cover everything involving the streamhandler in this class so you avoid the null pointer exception.